@xen-orchestra/proxy/src/app/_debounceResource.js

```diff
@@ -1,6 +1,6 @@
 import Disposable from 'promise-toolbox/Disposable'
 
-export async function debounceResource(pDisposable, hooks, delay = 0) {
+export async function debounceResource(pDisposable, addDisposeListener = dispose => Function.prototype, delay = 0) {
```

Comment:
Is `addDisposeListener` optional?
What about `delay`?

Optional params should go after mandatory ones.

Reply:
`delay` too. The delay default value is `0`.

Reply:
Why does it make sense for `delay` to be optional? :thinking:

Reply:
If we want to not debounce the resource. It simplify the usage of this function.

Reply:
Yes, it's either when the delay is computed dynamically (or from a config), and `debounce` already supports it, either via `undefined` or `0`.

But I see no use cases where the user would want to call `debounce` without passing a `delay` argument. :slightly_smiling_face: